uint8 gpio_id
uint8 edge_change_type
uint32 time_since_boot_us 